tests/src/test/scala/com/nvidia/spark/rapids/CostBasedOptimizerSuite.scala

```diff
@@ -57,20 +63,70 @@ class CostBasedOptimizerSuite extends SparkQueryCompareTestSuite
     assert(0.5d === MemoryCostHelper.calculateCost(GIGABYTE, 2) +- 0.01)
   }
 
+  test("Avoid transition to GPU for trivial projection after CPU SMJ") {
+    logError("Avoid transition to GPU for trivial projection after CPU SMJ")
+
+    val conf = DEFAULT_CONF
+      .set(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key, "true")
+      .set(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key, "-1")
+      .set(RapidsConf.ENABLE_CAST_STRING_TO_TIMESTAMP.key, "false")
+      .set(RapidsConf.ENABLE_REPLACE_SORTMERGEJOIN.key, "false")
+      .set(RapidsConf.TEST_ALLOWED_NONGPU.key,
```

Comment:
This mutates `DEFAULT_CONF` and risks leaking test-specific settings into other tests that use `DEFAULT_CONF`.

Reply:
`DEFAULT_CONF` is a `def` not a `val` so a new instance of `SQLConf` is created for each test. I can change this to a method to make this clearer though.

Reply:
oh my apologies for misreading it.  Having the name all caps does make it look like a `val`.